src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Internal.hs

```diff
@@ -66,27 +70,47 @@ data ArithmeticCircuit a i o = ArithmeticCircuit
         -- ^ The witness generation functions
         acRNG     :: StdGen,
         -- ^ random generator for generating unique variables
-        acOutput  :: o (Var i)
+        acOutput  :: o (Var a i)
         -- ^ The output variables
     } deriving (Generic)
-deriving instance (NFData a, NFData (o (Var i)), NFData (Rep i))
+deriving instance (NFData a, NFData (o (Var a i)), NFData (Rep i))
     => NFData (ArithmeticCircuit a i o)
 
-acInput :: Representable i => i (Var i)
-acInput = fmapRep InVar (tabulate id)
+acInput0 :: Representable i => i (SysVar i)
+acInput0 = fmapRep InVar (tabulate id)
 
-data Var i
+acInput :: Representable i => i (Var a i)
+acInput = fmapRep (SysVar . InVar) (tabulate id)
+
+------------------------------ Symbolic compiler context ----------------------------
+
+data SysVar i
```

Comment:
Maybe `Var` definitions should not be in the section "Symbolic compiler context", but next to the `data ArithmeticCircuit` declaration